src/components/FilterBar/FilterDisplay/Inputs/RelativeDateInput.react.js

```diff
@@ -4,49 +4,81 @@ export class RelativeDateInput extends React.Component {
   constructor(props) {
     super(props);
     this.state = { value: this.props.value || { from: null, to: null, value: null } };
+    this.setDisplayDates(this.props.value['value']);
   }
 
-  onRelativeChange(event) {
-    var optionElement = event.target.childNodes[event.target.selectedIndex];
-    var from = moment(parseInt(optionElement.getAttribute('data-from')));
-    var to = moment(parseInt(optionElement.getAttribute('data-to')));
-    var newValue = { value: optionElement.value, from: from.format(this.props.dateFormat), to: to.format(this.props.dateFormat) };
+  // If relative option selected, set dates for the datepickers to display
+  setDisplayDates(relativeDateSelection) {
+    if(relativeDateSelection === undefined || relativeDateSelection == '') {
+      return;
+    }
+
+    var selected            = this.relativeOptions()[relativeDateSelection];
+    this.state.displayFrom  = selected.from.format(this.props.dateFormat);
+    this.state.displayTo    = selected.to.format(this.props.dateFormat);
+  }
 
-    this.setState({ value: newValue });
+  onRelativeChange(event) {
+    var selectedOption = $(event.target.childNodes[event.target.selectedIndex]);
+    var newValue = { value: selectedOption.val() };
+    this.state = { value: newValue }
     this.updateFilter(newValue);
   }
 
+  onDatePickerChange(event) {
+    var newValue = {
```

Comment:
So we retain the previous values, whether they came from the relative selection or a custom date selection.